### DISCUSSION TITLE
How to know the object's key ?

### DISCUSSION THREAD

**Discussion by yaallco** [nlohmann/json#2469] [Q&A] [answered]
Hello,

When receiving a json object, how can I check if the first element from de JSON is "pi" ?
From your exemple :

json j2 = {

        {"pi", 3.141},
        {"happy", true},
        {"name", "Niels"},
        {"nothing", nullptr},
        {"answer", {
            {"everything", 42}
        }},
        {"list", {1, 0, 2}},
        {"object", {
            {"currency", "USD"},
            {"value", 42.99}
        }}
    };

Thanks

**Comment by nlohmann** [comment, marked as answer]
This is not possible. The value 3.141 has no connection to its parent object.

**Reply by yaallco** [reply]
Oh, I can't know wich is the first element from a json object ?
And how can I check if a specific key exists in the json object?

**Reply by nlohmann** [reply]
You can access the first element with front (https://nlohmann.github.io/json/api/basic_json/front/).

You can use contains (https://nlohmann.github.io/json/api/basic_json/contains/) to check if an object contains a key.

**Reply by yaallco** [reply]
I didn't see your api ! 
Thanks for the answers, i'm a beginner with c++ haha
